### ISSUE TITLE
Bouncycastle-support: Move some code to quarkus-security

### ISSUE THREAD

**Issue by JiriOndrusek** [apache/camel-quarkus#2922]
Quarkus-security extension registers bouncycastle provider and register some classes for reflection.

It makes sense to put more registrations there and remove it from bouncycastle-support.

(see https://github.com/apache/camel-quarkus/pull/2915#issuecomment-883336289)

**Comment by michalvavrik** [comment]
Could you please provide an example why these registrations https://github.com/apache/camel-quarkus/blob/main/extensions-support/bouncycastle/deployment/src/main/java/org/apache/camel/quarkus/support/bouncycastle/deployment/BouncyCastleSupportProcessor.java#L46 are needed (e.g. concrete IT that fails when the classes are not registered)? Also see https://github.com/quarkusio/quarkus/issues/29366. Such example would be very helpful for me and might help you with this issue as there is related Quarkus issue https://github.com/quarkusio/quarkus/issues/28250.

**Comment by ppalaga** [comment]
@JiriOndrusek who added the lines around https://github.com/apache/camel-quarkus/blob/main/extensions-support/bouncycastle/deployment/src/main/java/org/apache/camel/quarkus/support/bouncycastle/deployment/BouncyCastleSupportProcessor.java#L46  might know. But even if he does not, the easiest way to figure out reliably would be to remove it, send a PR and you'll see on the CI what is failing.

**Comment by JiriOndrusek** [comment]
I'm going to find the affected tests. Trying PR with a failing CI seems like an easy way to do it.

**Comment by michalvavrik** [comment]
Thanks for the PR, but the CI is green.

**Comment by JiriOndrusek** [comment]
I see... I would say that code is not necessary anymore. With te respect of the date of this issue, it might not happen anymore. I soo so,me code for example in [quarkus security](https://github.com/quarkusio/quarkus/blob/main/extensions/security/deployment/src/main/java/io/quarkus/security/deployment/SecurityProcessor.java#L170-L183). The code there was added after the creation of this issue, hence this issue is outdated now. I'll remove unnecessary registration and close the issue after that.

**Comment by michalvavrik** [comment]
Yes, I think the code in the quarkus security is also unnecessary, I'll check that integration test modules guarantee it works and suggest removal.
